modules/aioseop_sitemap.php

```diff
@@ -403,7 +446,175 @@ public function __construct() {
 			// TODO is this required for dynamic sitemap?
 			add_action( 'transition_post_status', array( $this, 'update_sitemap_from_posts' ), 10, 3 );
 			add_action( 'after_doing_aioseop_updates', array( $this, 'scan_sitemaps' ) );
+			add_action( 'after_doing_aioseop_updates', array( $this, 'upgrade_excluded_categories_to_excluded_terms' ) );
 			add_action( 'all_admin_notices', array( $this, 'sitemap_notices' ) );
+			add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_scripts' ), 20 );
+			add_action( "wp_ajax_{$this->prefix}", array( $this, 'ajax' ) );
+		}
+
+		/**
+		 * Migrate the values in "Excluded Categories" to "Excluded Terms".
+		 *
+		 * @since 3.0
+		 */
+		function upgrade_excluded_categories_to_excluded_terms() {
+			global $aioseop_options;
+			$aioseop_options = aioseop_get_options();
+			$options   = $aioseop_options['modules'][ "{$this->prefix}options" ];
+
+			if ( ! empty( $options[ "{$this->prefix}excl_categories" ] ) ) {
+				$options[ "{$this->prefix}excl_terms" ] = $options[ "{$this->prefix}excl_categories" ];
+				unset( $options[ "{$this->prefix}excl_categories" ] );
+				$aioseop_options['modules'][ "{$this->prefix}options" ] = $options;
+				update_option( 'aioseop_options', $aioseop_options );
+			}
+		}
+
+
+		/**
+		 * The single entry function to handle all ajax stuff.
+		 *
+		 * @since 3.0
+		 */
+		function ajax() {
+			check_ajax_referer( $this->prefix, 'nonce' );
+			$response = array();
+			switch ( $_POST['_action'] ) {
+				case 'fetch_terms':
+					$args = wp_parse_args( $_POST['taxonomy'] );
+					$taxonomies = $args['aiosp_sitemap_taxonomies'];
+					if ( empty( $taxonomies ) ) {
+						wp_send_json_error( array( 'msg' => __( 'No taxonomies selected', 'all-in-one-seo-pack' ) ) );
+					}
+
+					$all_terms = array();
+
+					foreach ( $taxonomies as $taxonomy ) {
+						$terms = get_terms(
+							array(
+								'taxonomy' => $taxonomy,
+								'fields' => 'id=>name',
+								'hide_empty' => false,
+								'name__like' => $_POST['term'],
```

Comment:
Curious, did you try using `search` at all?

Reply:
no but it seems to be doing the same thing

Reply:
You didn't test the comparison? The reason why I mention it, is because I believe the LIKE query is heavy on the database, and using `search` might be using a different SQL string.